File: numberOfFactors.js
// Gives  the number of factors number N has
function numberOfFactors(N) {
  let count = 0;
  for (let i = 0; i * i < N; i++) {
    if (N % i == 0) {
      if (N / i == i) {
        count++;
      } else {
        count += 2;
      }
    }
  }
  return count;
}

console.log(numberOfFactors(10));

File: Java/numberOfFactors.java
public class numberOfFactors{
    public static int factors (int N){
        int count = 0;
        if(N == 0) return 0;
        for(int i=1; i*i<=N; i++){
            if(N % i == 0){
                if(N/i == i){
                    count++;
                }else{
                    count+=2;
                }
            }
        }
        return count;
    }
    public static void main(String args[]){
        System.out.print(factors(49));
    }
}